flink-end-to-end-perf-tests/flink-basic-operations/src/main/java/org/apache/flink/basic/operations/PerformanceTestJob.java

```diff
@@ -0,0 +1,225 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.basic.operations;
+
+import org.apache.flink.api.common.ExecutionMode;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.api.java.typeutils.TupleTypeInfo;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.basic.utils.FromElements;
+import org.apache.flink.basic.utils.ValueStateFlatMap;
+import org.apache.flink.basic.utils.WordCountData;
+import org.apache.flink.contrib.streaming.state.RocksDBStateBackend;
+import org.apache.flink.core.fs.FileSystem.WriteMode;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.runtime.jobgraph.ScheduleMode;
+import org.apache.flink.runtime.state.filesystem.FsStateBackend;
+import org.apache.flink.streaming.api.CheckpointingMode;
+import org.apache.flink.streaming.api.TimeCharacteristic;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.co.CoProcessFunction;
+import org.apache.flink.streaming.api.graph.StreamGraph;
+import org.apache.flink.util.Collector;
+
+/**
+ * End-to-end perf test for end-to-end perf test.
+ *
+ * <p>The sources are generated and bounded. The result is always constant.
+ *
+ * <p>Parameters:
+ * --jobName job's name
+ * --parallelism  source Node's parallelism
+ * --checkPointMode  checkpointMode，atleastone or exactlyonce
+ * --seed
+ * --sleepNum
+ * --streamPartitioner
+ * --scheduleMode
+ * --executionMode
+ * --inputPath if source data in hdfs then input the inputPath
+ * --checkpointInterval
+ * --checkpointPath
+ * --stateBackend
+ * --checkpointTimeout
+ */
+public class PerformanceTestJob {
+	private static byte[] size10B = new byte[10];
+	private static String recordSize10B = new String(size10B);
+	private static byte[] size100B = new byte[100];
+	private static String recordSize100B = new String(size100B);
+	private static byte[] size1KB = new byte[1024];
+	private static String recordSize1KB = new String(size1KB);
+	private static String recordValue;
+
+	protected DataStream<Tuple2<String, String>> generateGraph(
+			String sourceName,
+			StreamExecutionEnvironment env,
+			int recordSize,
+			long maxCount,
+			int sleepNum,
+			int seed,
+			int parallism,
+			String streamPartitioner,
+			String outputPath){
+		if (recordSize == 10){
```

Comment:
there should be a space before the opening brace

Reply:
ok, i update it